json/src/value.rs

```diff
@@ -327,6 +395,86 @@ impl Value {
     }
 }
 
+/// A type that can be used to index into a `serde_json::Value`. See the `get`
+/// and `get_mut` methods of `Value`.
+pub trait Index: private::Sealed {
```

Comment:
I added an explanation in 7ded5a56a1e5b92ee554f48f7cdf116f08f3725e. If we discover a different idiomatic way, we can always change it.